src/tests/secp256k1_tests.rs

```diff
@@ -370,3 +373,47 @@ proptest::proptest! {
         assert!(pub_key_1.verify(message, &typed_sig).is_ok());
     }
 }
+
+#[test]
+fn wycheproof_test() {
+    let test_set = TestSet::load(EcdsaSecp256k1Sha256).unwrap();
+    for test_group in test_set.test_groups {
+        let pk = Secp256k1PublicKey::from_uncompressed(&test_group.key.key);
+        for test in test_group.tests {
+            let bytes = match Signature::from_der(&test.sig) {
+                Ok(s) => s.serialize_compact(),
+                Err(_) => {
+                    assert!(test.result == wycheproof::TestResult::Invalid);
+                    continue;
+                }
+            };
+
+            // Wycheproof tests do not provide a recovery id, iterate over all possible ones to verify.
+            let mut n_bytes = [0u8; 65];
+            n_bytes[..64].copy_from_slice(&bytes[..]);
+            let mut res = TestResult::Invalid;
+
+            for i in 0..4 {
+                n_bytes[64] = i;
+                let sig = <Secp256k1Signature as ToFromBytes>::from_bytes(&n_bytes).unwrap();
+                if pk
+                    .verify_hashed(&k256::sha2::Sha256::digest(&test.msg), &sig)
+                    .is_ok()
+                {
+                    res = TestResult::Valid;
```

Comment:
Ok, so if it least one pk succeeds we assume it's fine. I'm fine with that just because wycheproof tests do not support ecrecover.